src/daemon/plugin.c

```diff
@@ -1078,14 +1078,66 @@ static int plugin_compare_read_func(const void *arg0, const void *arg1) {
     return 0;
 } /* int plugin_compare_read_func */
 
+/*
+ * Set the time of the first read according to the settings AlignRead and
+ * Interval in collectd.conf.
+ *
+ * The AlignRead value is in seconds. The value modulo 60 determines the second
+ * of a minute. The integer part of the value divided by 60 determines the
+ * minute of an hour. Optionally, decimal places can be used to set the
+ * fraction of a second (up to milliseconds precision).
+ */
+static void plugin_set_first_read_time(read_func_t *rf) {
+
+  double start = rf->rf_ctx.align_read; /* (default is -1.0) */
+
+  /* if AlignRead is not set, start reading immediately */
+  if (start < 0) {
+    rf->rf_next_read = cdtime();
+  } else {
+    cdtime_t now = cdtime();
+
+    /* get minute in hour and second in minute and millisecond in second */
```

Comment:
Completely unclear comment.

It absolutely hard to understand what it means:  `get miliseconds since start of hour`. (And I'm still unclear if that is really this value).

Reply:
I replaced the comment some time ago.